test/thing-unit-tests.js

```diff
@@ -69,6 +69,70 @@ describe( "thing shadow class unit tests", function() {
       });
    });
 
+   describe( "register a thing shadow name", function() {
+//
+// Verify that the thing shadow invokes the register callback when subscription to all
+// topics are finished. The callback is invoked based on the callback from the mqtt library.
+//
+
+      var thingShadowsConfig = {
+         keyPath: 'test/data/private.pem.key',
+         certPath: 'test/data/certificate.pem.crt',
+         caPath: 'test/data/root-CA.crt',
+         clientId: 'dummy-client-1',
+         region: 'us-east-1'
+      };
+
+      it("should trigger error when a subscription fails", function () {
+
+        var stubTriggerError = sinon.stub(mockMQTTClient, 'triggerError', function(){return true;});
```

Comment:
Haven't used sinon that much before, so don't know if I did it the correct way :)